ferret/modeling/text_helpers.py

```diff
@@ -207,8 +205,22 @@ def _check_sample(self, text):
         ):
             raise ValueError("Input sample type is not supported")
 
-        if isinstance(text, str) or isinstance(text, tuple):
+        if isinstance(text, str):
             return [text]
+        # The SequenceClassificationHelper is only used for the text-classification and Natural Language Inference tasks.
+        # The following condition takes care of the NLI task (which was causing problems) in the SHAP explainer. 
+        # The expected input for the NLI task is constructed as follows:
+
+        # >>> premise = "I first thought that I liked the movie, but upon second thought it was actually disappointing."
+        # >>> hypothesis = "The movie was good."
+        # >>> sample = (premise, hypothesis)
+
+        # where the tuple given by "sample" is indeed the expected text input to the explainer.
+        # As currently designed, the SHAP explainer expects the input instead to be either a string or a list with a single string, 
+        # and *not* a list of a tuple. We her construct the text input for the explainers as "premise [SEP] hypothesis"
+        elif isinstance(text, tuple) and len(text) == 2:
+            if isinstance(text[0], str) and isinstance(text[1], str):
+                return [text[0] + " [SEP] " + text[1]]
```

Comment:
I am pretty sure not all encoder models use `[SEP]`, e.g., BERT does but I'm not sure distilbert/roberta, for example, do the same. Maybe we can check whether the used tokenizer has that special token (e.g., `tokenizer.sep_token`?) and, if it doesn't, we could go with the hardcoded "[SEP]" (again, printing a message to say that).

Reply:
Yes, you are absolutely right. I'll implement that.